dnf-json

```diff
@@ -123,98 +126,147 @@ def repo_checksums(base):
+                        "license": package.license
+                    })
+                self.exit_with_success({
+                    "checksums": repo_checksums(base),
+                    "packages": packages
+                })
+
+            elif command == "depsolve":
+                log.info("start depsolve")
+                errors = []
+
+                try:
+                    base.install_specs(
+                        arguments["package-specs"],
+                        exclude=arguments.get("exclude-specs", [])
+                    )
+                except dnf.exceptions.MarkingErrors as e:
+                    log.info("error install_specs")
+                    return self.exit_with_dnf_error(
+                        "MarkingErrors",
+                        f"Error occurred when marking packages for installation: {e}"
+                    )
+
+                try:
+                    base.resolve()
+                except dnf.exceptions.DepsolveError as e:
+                    log.info("error depsolve")
+                    return self.exit_with_dnf_error(
+                        "DepsolveError",
+                        (
+                            "There was a problem depsolving "
+                            f"{arguments['package-specs']}: {e}"
+                        )
+                    )
+
+                dependencies = []
+                for tsi in base.transaction:
+                    # Avoid using the install_set() helper, as it does not guarantee
+                    # a stable order
+                    if tsi.action not in dnf.transaction.FORWARD_ACTIONS:
+                        continue
+                    package = tsi.pkg
+
+                    dependencies.append({
+                        "name": package.name,
+                        "epoch": package.epoch,
+                        "version": package.version,
+                        "release": package.release,
+                        "arch": package.arch,
+                        "repo_id": package.reponame,
+                        "path": package.relativepath,
+                        "remote_location": package.remote_location(),
+                        "checksum": (
+                            f"{hawkey.chksum_name(package.chksum[0])}:"
+                            f"{package.chksum[1].hex()}"
+                        )
+                    })
+                self.exit_with_success({
+                    "checksums": repo_checksums(base),
+                    "dependencies": dependencies
+                })
+
+log.info("Starting the dnf-json server")
+
+SOCKET_FILE_PATH = "/run/dnf-json.sock"
+
+LISTEN_FDS = int(os.environ.get("LISTEN_FDS", 0))
+LISTEN_PID = os.environ.get("LISTEN_PID", None) or os.getpid()
+
+log.debug("env %d and %d", LISTEN_FDS, LISTEN_PID)
+
+class SystemDActivationSocketServer(socketserver.UnixStreamServer):
+    def server_bind(self):
+        log.debug("service bind")
+        if LISTEN_FDS == 0:
```

Comment:
I think we may want to only support LISTEN_FDS=1 so we avoid confusion due to both this dameon and systemd "owning" the same socket.

Reply:
In the end we need to have this thing. Because sometime we want to be able to start dnf-json without systemd. This is the case when running in a container and for running the test cases generation.